core/src/main/java/org/apache/iceberg/rest/requests/CreateNamespaceRequest.java

```diff
@@ -19,80 +19,24 @@
 package org.apache.iceberg.rest.requests;
 
 import java.util.Map;
-import java.util.Objects;
 import org.apache.iceberg.catalog.Namespace;
-import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
-import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
-import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.apache.iceberg.rest.RESTRequest;
+import org.immutables.value.Value;
 
 /** A REST request to create a namespace, with an optional set of properties. */
-public class CreateNamespaceRequest implements RESTRequest {
+@Value.Immutable
+public interface CreateNamespaceRequest extends RESTRequest {
```

Comment:
This class has been released, so this is a breaking API change. We can extract an interface with a different name and have the Immutable class use this name maybe?

Reply:
I don't think we're breaking the API. All the places that have been using `CreateNamespaceRequest` should work as before. We also maintain backwards-compatibility through the original `Builder` (which is deprecated now) and it's being used in https://github.com/apache/iceberg/blob/772fcecf17467b1ba1dc9cf30391203fc02f575c/core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java#L333-L334.

The only thing that would break is probably if somebody was using the public no-arg constructor directly (which only existed for Jackson) or maybe if somebody was subclassing `CreateNamespaceRequest`

Reply:
I'm talking about binary compatibility, not source compatibility. Changing a class to an interface breaks binary compatibility.

Reply:
I've fixed the ABI issue by introducing a `NamespaceCreateRequest` and deprecating the `CreateNamespaceRequest`